widget/entry.go

```diff
@@ -296,25 +312,26 @@ func (e *Entry) SetReadOnly(ro bool) {
 
 // SetText manually sets the text of the Entry to the given text value.
 func (e *Entry) SetText(text string) {
-	e.textProvider().SetText(text)
+	e.textProvider().setText(text)
 	e.updateText(text)
 
 	if text == "" {
-		e.Lock()
-		e.CursorColumn = 0
-		e.CursorRow = 0
-		e.Unlock()
-		e.Refresh()
-	} else {
-		provider := e.textProvider()
+		e.SetFieldsAndRefresh(func() {
+			e.CursorColumn = 0
+			e.CursorRow = 0
+		})
+		return
+	}
+	provider := e.textProvider()
```

Comment:
Would it be safer to move this inside SetFields so it is called while holding the lock?